inugami_webapp/src/main/webapp/js/app/components/forms/input.switch.ts

```diff
@@ -0,0 +1,185 @@
+
+    @Input() validator              : any;
+
+    @Output() onFocus               : EventEmitter<any> = new EventEmitter();
+
+    @Output() onChange              : EventEmitter<any> = new EventEmitter();
+
+    onModelChange                   : Function          = () => { };
+
+    onModelTouched                  : Function          = () => { };
+
+    displayedLabel                  : string;
+
+    styleClasses                    : string[];
+
+    focused                         : boolean;
+
+    checked                         : boolean           = false;
+
+    valid                           : boolean           = true;
+
+    enableValidator                 : boolean           = false;
+
+    /*****************************************************************************
+    * INIT
+    *****************************************************************************/
+
+    ngOnInit() {
+        this.checked = true;
+        this.styleClasses = []
+        this.displayedLabel = this.labelOn;
+    }
+    constructor(private cd: ChangeDetectorRef) { }
+
+    /****************** ***********************************************************
+     * IMPLEMENTS ControlValueAccessor
+     *****************************************************************************/
+
+
+    writeValue(checked: any): void {
+        this.checked = checked;
+        this.displayedLabel = checked ? this.labelOn : this.labelOff;
+        this.cd.markForCheck();
+    }
+
+    registerOnChange(fn: Function): void {
+        this.onModelChange = fn;
+    }
+
+    registerOnTouched(fn: Function): void {
+        this.onModelTouched = fn;
+    }
+    setDisabledState(val: boolean): void {
+        this.disabled = val;
+    }
+
+    /*****************************************************************************
+    * METHODS
+    *****************************************************************************/
+
+
+    toggle(event: Event) {
+        this.updateModel(event, !this.checked);
+    }
+
+    updateModel(event: Event, value: boolean) {
+        this.checked = value;
+        this.processValidator();
+        this.displayedLabel = value ? this.labelOn : this.labelOff;
+        this.onModelChange(this.checked);
+        this.onChange.emit({
+            originalEvent: event,
+            checked: this.checked
+        });
+    }
+
+    getBaseStyleClass(){
+        let result= ["inugami-input-switch-button"];
+        if(!this.valid){
+            result.push("validation-error");
+        }
+        if(this.checked){
+            result.push("state-on");
+        }else{
+            result.push("state-off");
+        }
+        return result.join(" ");
+    }
+    /*****************************************************************************
+     * HANDLERS
+     *****************************************************************************/
+    onCompoFocused($event) {
+        if (this.focused === false) {
+            this.onFocus.emit({
```

Comment:
When you fire event, you should create a typescript model for help integration of your component in specific view or plugins